src/textual/message_pump.py

```diff
@@ -243,7 +243,7 @@ def is_attached(self) -> bool:
         except NoActiveAppError:
             return False
         node: MessagePump | None = self
-        while (node := node._parent) is not None:
+        while node is not None and (node := node._parent) is not None:
```

Comment:
This adds an additional check per loop, which is never going to be truthy. I think we should find another way to satisfy Mypy if that is why this was added.